lib/utils/emitWarning.mjs

```diff
@@ -42,24 +48,24 @@ function emitWarning(message, type, codeKey, detail) {
  * Use this function for deprecation warnings, instead of `process.emitWarning()`.
  *
  * @param {string} message
- * @param {keyof CODES} codeKey
+ * @param {keyof typeof DEPRECATION_CODES} codeKey
```

Comment:
question

Do you really need the `typeof`?

Reply:
Yes, it’s necessary. The code runs without `typeof` in JSDoc, but omitting it breaks IDE rename (see https://github.com/microsoft/TypeScript/issues/51889)

Reply:
IMHO this bug doesn't matter, if that's the only reason you can get rid of it.

Reply:
I also don't think the issue matters (edge case). Instead, I'm concerned that `keyof typeof` might confuse us. If you don't find other issues, can you remove `typeof`?

Reply:
Thanks to both of you for the comments!
Fixed via 7125ee6cdd04654d55aeee8c1e9509e63631ad97